ci/build-fips-action/entrypoint.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env sh
+
+git config --global --add safe.directory /github/workspace
+
+# Install Go
+url="https://go.dev/dl/go${GO_VERSION}.linux-${TARGETARCH}.tar.gz"
+echo "Downloading ${url}"
+curl -Lo go.tar.gz "$url"
+tar -zxvf go.tar.gz -C /usr/local
+export PATH="/usr/local/go/bin:${PATH}"
+
+# Install builder
+cd otelcolbuilder || exit 1
+mkdir "${HOME}/bin"
+export PATH="${HOME}/bin:${PATH}"
+make install-builder
```

Comment:
These install commands seem like they'd be more at home in the Dockerfile.

Reply:
I couldn't find a way of passing the Go version into the Dockerfile itself from GitHub Actions.

Reply:
I moved the Go installation steps to the Dockerfile and hardcoded the Go version for now. I was hoping to minimize the amount of places where we need to bump the Go version but it does cut down on runtimes.

I don't think it's possible to use anything except the `ci/build-fips-action` directory as the context when GitHub Actions builds the Docker image. I've left the builder installation in `entrypoint.sh` as a result.